### DISCUSSION TITLE
The most efficient way to delete old records?

### DISCUSSION THREAD

**Discussion by lileishihaoren** [litedb-org/LiteDB#2026] [Q&A]
using (var db = new LiteDatabase(_fileName))
            {
                var sql = $"SELECT $._id FROM {colName} order by $._id desc offset {maxKeepNum}";
                var list = db.Execute(sql).ToList();
                if (list.Any())
                {
                    var intList = list.Select(x => x["_id"].AsInt32).ToList();
                    var col = db.GetCollection<MyClass>(colName);
                    foreach (var i in intList)
                    {
                        col.Delete(i);
                    }
                }
            }
I use the code to delete some old records,is there any more efficient way? Thanks. 
in LiteDB V5

**Comment by lbnascimento** [comment]
@lileishihaoren You can run all of your `Delete` operations inside a single transaction:

```C#
db.BeginTrans();
foreach (var i in intList)
{
    col.Delete(i);
}
db.Commit();
```

This has the (potentially unwanted) side effect that, if a single operation fails, all of the operations are discarded. However, this is the fastest way to run batch operations.
